particle-php/Filter
# Add Remove and RemoveNull filter rules

### What

Removing values from the input array requires some core refactoring.
### How to test
1. See that there are tests that can remove values form the to filter array
2. See that the test coverage is still 100% and the quality a 10
### ToDo
- [x] Find a way to properly unset values from the data container
- [x] Add decent tests for the removeNull filter rule
- [x] Fix removal on multidimensional arrays
- [x] Extend readme example
- [x] ~~Code quality 10~~ Make it as clean as possible
### Ticket

 #42
